skbio/tree/_tree.py

```diff
@@ -3098,3 +3098,33 @@ def shuffle(self, k=None, names=None, shuffle_f=np.random.shuffle, n=1):
 
             yield self
             counter += 1
+
+    def _cache_leafcount(self):
+        """ Counts the number of leaves under each subtree."""
+        for n in self.postorder():
```

Comment:
To protect against future API changes, and `include_self` be explicitly passed here and in the other two methods?